Add boolean field type

Ariana-B opened the pull request:
### Reason for this pull request

See #1937 


### Proposed changes

- Introduce `BoolDocField` to enable boolean type properties in eo3 metadata 



 - [x] Closes #1937 
 - [x] Tests added / passed
 - [x] Fully documented, including `docs/about/whats_new.rst` for all changes

<!--
See https://github.com/blog/2111-issue-and-pull-request-templates for more
information on pull request templates.

-->


<!-- readthedocs-preview opendatacube start -->
----
📚 Documentation preview 📚: https://opendatacube--1949.org.readthedocs.build/en/1949/

<!-- readthedocs-preview opendatacube end -->

Commit:
archive less mature works even if ds is already indexed

Commit:
update whats_new

Commit:
spelling

Commit:
add 'has' logic to mockdb

Commit:
add boolean type field

Commit:
add boolean type field

pjonsson commented on datacube/drivers/postgres/_fields.py:
```diff
@@ -260,6 +260,22 @@ def parse_value(self, value):
+        return cast(value, postgres.BOOLEAN)
+
+    @override
+    def parse_value(self, value):
```
Can this one also get a return type like the postgis method?

Commit:
update model fields and update whats_new

Commit:
update model fields and update whats_new

Commit:
add return types to parse_value methods

Commit:
add return types to parse_value methods

Ariana-B committed:
Merge branch 'develop' into eo3_bool_support

Commit:
linting

Commit:
linting

pjonsson committed:
Fix output_crs type annotation

This should have been a part of #1940.

SpacemanPaul committed:
Merge branch 'develop' into always_run_alm

SpacemanPaul committed:
Merge pull request #1948 from opendatacube/always_run_alm

archive-less-mature works even for indexed datasets

SpacemanPaul committed:
Merge branch 'develop' into outputcrs-type

pjonsson committed:
Remove types from doc strings

These are injected from the type annotation
by Sphinx.

Refs #1869

SpacemanPaul committed:
Merge pull request #1950 from pjonsson/outputcrs-type

Fix output_crs type annotation

SpacemanPaul reviewed (changes requested):
Most of the way there.  You'll also need to link the boolean search field type to the numeric search index table in the [search_field_map](https://github.com/opendatacube/datacube-core/blob/76ba30fbbd9301c64942f1378d5ebcd649461946/datacube/drivers/postgis/_schema.py#L370)

Ideally we would add an indexed boolean search field to a test to make sure the int/bool conversion happens correctly at indexing and in search queries.

(postgres driver should just work with what you have)

pjonsson committed:
tests: fix type signatures

pjonsson committed:
alembic: document autogeneration

Add an example of how to autogenerate
the alembic upgrade.

pjonsson committed:
alembic: do not generate empty upgrades

Only generate an upgrade file when using
--autogenerate if there is a difference
to the database.

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/astral-sh/ruff-pre-commit: v0.11.12 → v0.11.13](https://github.com/astral-sh/ruff-pre-commit/compare/v0.11.12...v0.11.13)
- [github.com/astral-sh/uv-pre-commit: 0.7.9 → 0.7.12](https://github.com/astral-sh/uv-pre-commit/compare/0.7.9...0.7.12)

dependabot[bot] committed:
build(deps): bump the python-deps group with 11 updates

Bumps the python-deps group with 11 updates:

| Package | From | To |
| --- | --- | --- |
| [boto3](https://github.com/boto/boto3) | `1.38.29` | `1.38.33` |
| [boto3-stubs](https://github.com/youtype/mypy_boto3_builder) | `1.38.29` | `1.38.33` |
| [botocore](https://github.com/boto/botocore) | `1.38.29` | `1.38.33` |
| [botocore-stubs](https://github.com/youtype/botocore-stubs) | `1.38.29` | `1.38.30` |
| [greenlet](https://github.com/python-greenlet/greenlet) | `3.2.2` | `3.2.3` |
| [hypothesis](https://github.com/HypothesisWorks/hypothesis) | `6.135.0` | `6.135.4` |
| [pandas](https://github.com/pandas-dev/pandas) | `2.2.3` | `2.3.0` |
| [requests](https://github.com/psf/requests) | `2.32.3` | `2.32.4` |
| ruamel-yaml | `0.18.12` | `0.18.14` |
| [ruff](https://github.com/astral-sh/ruff) | `0.11.12` | `0.11.13` |
| [tomlkit](https://github.com/sdispater/tomlkit) | `0.13.2` | `0.13.3` |


Updates `boto3` from 1.38.29 to 1.38.33
- [Release notes](https://github.com/boto/boto3/releases)
- [Commits](https://github.com/boto/boto3/compare/1.38.29...1.38.33)

Updates `boto3-stubs` from 1.38.29 to 1.38.33
- [Release notes](https://github.com/youtype/mypy_boto3_builder/releases)
- [Commits](https://github.com/youtype/mypy_boto3_builder/commits)

Updates `botocore` from 1.38.29 to 1.38.33
- [Commits](https://github.com/boto/botocore/compare/1.38.29...1.38.33)

Updates `botocore-stubs` from 1.38.29 to 1.38.30
- [Release notes](https://github.com/youtype/botocore-stubs/releases)
- [Commits](https://github.com/youtype/botocore-stubs/commits)

Updates `greenlet` from 3.2.2 to 3.2.3
- [Changelog](https://github.com/python-greenlet/greenlet/blob/master/CHANGES.rst)
- [Commits](https://github.com/python-greenlet/greenlet/compare/3.2.2...3.2.3)

Updates `hypothesis` from 6.135.0 to 6.135.4
- [Release notes](https://github.com/HypothesisWorks/hypothesis/releases)
- [Commits](https://github.com/HypothesisWorks/hypothesis/compare/hypothesis-python-6.135.0...hypothesis-python-6.135.4)

Updates `pandas` from 2.2.3 to 2.3.0
- [Release notes](https://github.com/pandas-dev/pandas/releases)
- [Commits](https://github.com/pandas-dev/pandas/compare/v2.2.3...v2.3.0)

Updates `requests` from 2.32.3 to 2.32.4
- [Release notes](https://github.com/psf/requests/releases)
- [Changelog](https://github.com/psf/requests/blob/main/HISTORY.md)
- [Commits](https://github.com/psf/requests/compare/v2.32.3...v2.32.4)

Updates `ruamel-yaml` from 0.18.12 to 0.18.14

Updates `ruff` from 0.11.12 to 0.11.13
- [Release notes](https://github.com/astral-sh/ruff/releases)
- [Changelog](https://github.com/astral-sh/ruff/blob/main/CHANGELOG.md)
- [Commits](https://github.com/astral-sh/ruff/compare/0.11.12...0.11.13)

Updates `tomlkit` from 0.13.2 to 0.13.3
- [Release notes](https://github.com/sdispater/tomlkit/releases)
- [Changelog](https://github.com/python-poetry/tomlkit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/sdispater/tomlkit/compare/0.13.2...0.13.3)

---
updated-dependencies:
- dependency-name: boto3
  dependency-version: 1.38.33
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: python-deps
- dependency-name: boto3-stubs
  dependency-version: 1.38.33
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: python-deps
- dependency-name: botocore
  dependency-version: 1.38.33
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: python-deps
- dependency-name: botocore-stubs
  dependency-version: 1.38.30
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: python-deps
- dependency-name: greenlet
  dependency-version: 3.2.3
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: python-deps
- dependency-name: hypothesis
  dependency-version: 6.135.4
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: python-deps
- dependency-name: pandas
  dependency-version: 2.3.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: python-deps
- dependency-name: requests
  dependency-version: 2.32.4
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: python-deps
- dependency-name: ruamel-yaml
  dependency-version: 0.18.14
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: python-deps
- dependency-name: ruff
  dependency-version: 0.11.13
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: python-deps
- dependency-name: tomlkit
  dependency-version: 0.13.3
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: python-deps
...

Signed-off-by: dependabot[bot] <support@github.com>

SpacemanPaul committed:
Merge pull request #1951 from pjonsson/remove-docstring-types

Remove types from doc strings

SpacemanPaul committed:
Merge branch 'develop' into alembic-no-empty-upgrades

SpacemanPaul committed:
Merge pull request #1956 from opendatacube/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

SpacemanPaul committed:
Merge pull request #1957 from opendatacube/dependabot/uv/python-deps-72af3924e5

build(deps): bump the python-deps group with 11 updates

SpacemanPaul committed:
Merge pull request #1955 from pjonsson/alembic-no-empty-upgrades

Document alembic upgrades

SpacemanPaul committed:
Expose Grid Workflow classes from datacube.api.

SpacemanPaul committed:
Update PR#

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pjonsson committed:
postgis: remove ALL_STATIC_TABLES

This is handled by include_name()
in env.py.

pjonsson committed:
postgis: fix alembic metadata object

There was even commented out code for
this, but I still missed this when switching
to inheriting DeclarativeBase.

pjonsson committed:
postgis: re-use session object

Continue to use the already existing
session object instead of creating
a new connection.

pjonsson committed:
postgis: stop parsing table names

There is an integer available further
up the call chain, so pass that along
instead of repeatedly parsing
the tablename and adding ignores for
MyPy.

Commit:
update postgis search_field_map

Commit:
update postgis search_field_map

Commit:
add s1 dataset with boolean

Commit:
add s1 dataset with boolean

SpacemanPaul committed:
Merge pull request #1958 from opendatacube/expose-grid-workflow-from-api

Expose Grid Workflow classes from datacube.api.

SpacemanPaul committed:
Merge branch 'develop' into remove-table-list

SpacemanPaul committed:
Merge pull request #1960 from pjonsson/remove-table-list

postgis: fix alembic metadata object

SpacemanPaul committed:
Merge branch 'develop' into postgis-pass-id

SpacemanPaul committed:
Merge pull request #1961 from pjonsson/postgis-pass-id

postgis: stop parsing table names

Commit:
serialize canonical_name in Measurement

Commit:
update test and whatsnew

SpacemanPaul committed:
Merge pull request #1962 from opendatacube/fix_measurement_serialize

serialize with `canonical_name` in `Measurement`

Commit:
update docs and add test

Commit:
update docs and add test

SpacemanPaul committed:
Prevent Alembic INFO log messages from breaking tests.

SpacemanPaul committed:
Fix postgis implementation of boolean search fields.

SpacemanPaul committed:
Merge remote-tracking branch 'origin/eo3_bool_support' into eo3_bool_support

# Conflicts:
#	datacube/drivers/postgis/_fields.py
#	docs/about/whats_new.rst

SpacemanPaul committed:
Update uv.lock.
